app/grandchallenge/components/signals.py

```diff
@@ -0,0 +1,13 @@
+from django.db.models.signals import pre_delete
+from django.dispatch import receiver
+
+from grandchallenge.algorithms.models import AlgorithmImage, AlgorithmModel
+from grandchallenge.evaluation.models import EvaluationGroundTruth, Method
+
+
+@receiver(pre_delete, sender=AlgorithmImage)
+@receiver(pre_delete, sender=AlgorithmModel)
+@receiver(pre_delete, sender=EvaluationGroundTruth)
+@receiver(pre_delete, sender=Method)
+def delete_file_from_s3(instance, **_):
+    instance._client.delete_object(Bucket=instance.bucket, Key=instance.key)

```

Comment:
Why go through the client and not call `instance.model.delete()`?

Reply:
Oh I didn't know that worked. That's much simpler.